pkg/controller/operands/kubevirt_test.go

```diff
@@ -247,6 +257,205 @@ var _ = Describe("KubeVirt Operand", func() {
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &corev1.ConfigMap{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
+
+				By("KV CR should contain the enabled feature gate", func() {
+					Expect(foundResource.Data[virtconfig.FeatureGatesKey]).Should(Equal(constFeatureGates + "," + fgEnabled))
+				})
+			})
+
+			It("should handle feature gates on update", func() {
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				existingResource.Data[virtconfig.FeatureGatesKey] = fmt.Sprintf("%s,%s,%s,%s", constFeatureGates, fgMissing, fgDisabled, fgNoChange)
+
+				hco.Spec.FeatureGates = map[string]bool{
+					fgEnabled:  true,
+					fgDisabled: false,
+					fgNoChange: true,
+				}
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := (*genericOperand)(newKvConfigHandler(cl, commonTestUtils.GetScheme()))
+				res := handler.ensure(req)
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &corev1.ConfigMap{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
+
+				By("Should add enabled FGs, remove missing FGs, remove disabled FGs and not change existing enabled FGs", func() {
+					found := foundResource.Data[virtconfig.FeatureGatesKey]
+					Expect(strings.Contains(found, fgEnabled)).To(BeTrue())
+					Expect(strings.Contains(found, fgMissing)).To(BeFalse())
+					Expect(strings.Contains(found, fgDisabled)).To(BeFalse())
+					Expect(strings.Contains(found, fgNoChange)).To(BeTrue())
+				})
+			})
+
+			It("should remove all KV feature gates if there are no managed KV feature gates in HC", func() {
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				existingResource.Data[virtconfig.FeatureGatesKey] = fmt.Sprintf("%s,%s,%s,%s", constFeatureGates, fgMissing, fgDisabled, fgNoChange)
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := (*genericOperand)(newKvConfigHandler(cl, commonTestUtils.GetScheme()))
+				res := handler.ensure(req)
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &corev1.ConfigMap{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
+
+				By("KV CR should not contain the WithHostPassthroughCPU feature gate", func() {
```

Comment:
I think the description is a little bit misleading since I don't see any reference to `WithHostPassthroughCPU` in this test case. Maybe the aim was to check that only the default const feature gate list of KV exists, and everything else that is not managed by the HCO CR will be deleted from the list

Reply:
typo. will fix